cli/compose/schema/data/config_schema_v3.1.json

```diff
@@ -167,7 +167,19 @@
         "ports": {
           "type": "array",
           "items": {
-            "type": ["string", "number"],
+            "oneOf": [
+              {"type": "number"},
+              {"type": "string"},
+              {
+                "type": "object",
+                "properties": {
+                  "mode": {"type": "string"},
+                  "target": {"type": "integer"},
+                  "published": {"type": "integer"},
+                  "protocol": {"type": "string"}
+                }
```

Comment:
Oh, I guess if it's not being used by swarmkit, we shouldn't add it. It's strange that it's in the API types if it's not used.